core/app/models/spree/variant.rb

```diff
@@ -80,6 +80,11 @@ class Variant < Spree::Base
 
     after_destroy :destroy_option_values_variants
 
+    scope :template_variants, -> do
```

Comment:
Not sure about this name, because it requires some brain work to understand what it refers to. What about something like `master_with_option_types`?

Reply:
I think `template_variants` is actually very clear (it tells the programmer what these things are) whereas `master_with_option_types` is only clear to people with a lot of Solidus experience. A new user would have to figure out why a variant can be some kind of `master` - does it have power over the others? And they would have to find out what `with_option_types` means. Then they would realize the method checks for `option_values` and... they'd be more confused by the naming of that. :D

Reply:
We (@tvdeyen and I) are settling in a 1:1 on `non_template_variants`, as `sellable` has more implications. It would also need prices, and option values.

Reply:
Yeah, my point is that a `template` is already a thing in people's heads, especially those without Solidus knowledge. 🤔 

I hear your points though, and they make sense, but I'm not sure I'm sold on it. I think there might be a better name, I'll try to come up with something. 

Also, I just realized this is something already kinda touched by the [has many association between a product and its variants](https://github.com/solidusio/solidus/blob/2f05527b1acdc0bca2909e1e04ec5f28a092e770/core/app/models/spree/product.rb#L42), but I'm sure there's a difference I'm missing. In which context should we use the one in your proposal vs the scoping already present in the association?

Reply:
Products without option types directly sell their "master" variant, and in that case it's not a template, but an actual variant.

Reply:
And e.g. in the backend variant select, we want the master variant of products without option types, but not the master variant of products with option types. Those can usually not be sold (think a T-shirt without size and color).